Nourleyy/post service

Nourleyy opened the pull request:
يادوبك الواحد يقوم يصلي الفجر 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced endpoints for creating, updating, deleting, and retrieving posts, including support for pagination and filtering by seller.
  - Added support for associating assets with posts, including validation and view models for post assets.
  - Posts now include additional details such as category, subcategory, location, condition, status, and seller information.

- **Bug Fixes**
  - Improved validation for post creation and updates, ensuring required fields and valid data.

- **Chores**
  - Updated token expiration durations for improved session management.
  - Enhanced dependency injection with new service registrations.

- **Refactor**
  - Removed unused properties and streamlined entity configurations for posts and post assets.
  - Cleaned up database seeding logic to prevent unnecessary data deletion.

- **Revert**
  - Removed previous database migration files and model snapshots to reset or update the schema.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Nourleyy committed:
Implemented Post Service and related DTOs

Nourleyy committed:
implemented posts Endpoint

gitnasr commented:
مخلصه التاسك علي الفجر عشان تقومي تدعي عليا😡

@Nourleyy

gitnasr commented on Dentizone.Application/DTOs/PostDTO/UpdatePostDTO.cs:
```diff
@@ -0,0 +1,21 @@
+{
+    public class UpdatePostDTO
+    {
+        public string Title { get; set; }
+        public string Description { get; set; }
+        public decimal Price { get; set; }
+        public PostItemCondition Condition { get; set; }
+        public string Street { get; set; }
+        public string City { get; set; }
+        public DateTime? ExpireDate { get; set; }
```
TODO: Check the FluentValidation

gitnasr commented on Dentizone.Application/DTOs/PostDTO/PostAssetDTO.cs:
```diff
@@ -0,0 +1,23 @@
+    {
+        public PostAssetDTOValidator()
+        {
+            RuleFor(x => x.Id)
+                .NotEmpty().WithMessage("Asset ID is required.");
+        }
```
Refactor NEeded

gitnasr commented on Dentizone.Application/Services/PostService.cs:
```diff
@@ -0,0 +1,83 @@
+            return _mapper.Map<List<PostViewDTO>>(post);
+        }
+
+        public async Task<PostViewDTO> UpdatePost(string postId, UpdatePostDTO updatePostDTO)
+        {
+            var post = _mapper.Map<Domain.Entity.Post>(updatePostDTO);
+            post.Id = postId;
+            var updatedPost = await _repo.UpdateAsync(post);
+
+            if (updatedPost == null)
+            {
+                throw new NotFoundException("Post not found");
+            }
+            return _mapper.Map<PostViewDTO>(updatedPost);
```
To be discussed with nourleyy

gitnasr commented on Dentizone.Application/DTOs/PostDTO/CreatePostDTO.cs:
```diff
@@ -0,0 +1,53 @@
+                .NotEmpty().WithMessage("Display order is required.");
+                
+            RuleFor(x => x.ExpireDate)
+                .GreaterThan(DateTime.Now)
```
@Nourleyy معلومه مفيده جديده

gitnasr committed:
hotfixes :

Add Post feature with DTOs and service implementation

This commit introduces a new `Post` feature in the application, including:
- New DTOs for creating, updating, and viewing posts.
- Implementation of the `IPostService` interface and `PostService` class.
- Corrections to property names and naming conventions in DTOs.
- Registration of `IPostRepository` for data access.
- Updates to `PostConfiguration` for entity configuration.
- Enhancements to `PostsController` for handling HTTP requests related to posts.

These changes improve the application's functionality and maintain consistency in code structure.

gitnasr committed:
Refactor post-related services and token management

- Rewrite `PostProfile.cs` to define mapping profiles for DTOs.
- Remove `SellerId` from `CreatePostDto` in `CreatePostDTO.cs`.
- Update `IPostService` to include `userId` in `CreatePost`.
- Increase access and refresh token durations in `TokenService.cs`.
- Modify `PostService` to set `SellerId` using `userId`.
- Refactor `PostsController` to inject `postService` directly and update method signatures.
- Enhance error handling in `CreatePost` method to check for valid `userId`.

gitnasr committed:
Add asset management features and update models

- Introduced `IAssetService` and `AssetService` for asset handling.
- Enhanced `UpdatePostDTO` with new properties: `Status`, `AssetIds`, and `ExpireDate`.
- Updated `PostService` to include asset validation and association logic.
- Set default value for `Status` in the `Post` entity.
- Refactored `PostAsset` for improved formatting.
- Updated `IPostRepository` with necessary methods.
- Registered `IPostAssetRepository` and `IAssetRepository` in DI.
- Added new migration files to reflect database schema changes.
- Updated `AppDbContextModelSnapshot` to match current model state.
- Modified configuration classes for `PostAsset` and `Post`.
- Improved `CatalogSeeder` to ensure proper seeding of categories and subcategories.

gitnasr committed:
Refactor asset handling in post DTOs

Removed PostAssetDto and its validator. Updated PostViewDTO to use PostAssetView instead of PostAssetDto for the Assets property. Introduced PostAssetView class to encapsulate asset details.

gitnasr committed:
Refactor PostAsset creation in PostService.cs

Reformatted the initialization of the `PostAsset` object for better structure. Removed the line that awaited the creation of `postAsset` within the loop, indicating a change in how assets are associated with posts, potentially allowing for batch processing.

gitnasr reviewed (commented):
Approved. Edits needed in the feature
